src/ir/opt/cfold.rs

```diff
@@ -244,6 +244,29 @@ pub fn fold_cache(node: &Term, cache: &mut TermCache<Term>) -> Term {
                     b.width() + w,
                 ))))
             }),
+            Op::Map(op) => match (get(0).as_array_opt(), get(1).as_array_opt()) {
+                (Some(a), Some(b)) => {
+                    // TODO: extend for n-ary arrays
+                    let mut res = a.clone();
```

Comment:
Why clone `a` here? Just create a new array. There's no need to copy all of the data inside of `a` only to throw it away a few lines later. Why not something like:

```suggestion
                    let mut res = Array::new(a.key_sort.clone(), a.default.clone(), Default::default(), a.size);
```